2/chapitre_02.md

```diff
@@ -0,0 +1,197 @@
+# Résumé
+
+Le but du chapitre est de se familiariser avec le pseudo-code utilisé dans le livre et avec les
+notations liées à la complexité algorithmique. Les deux principaux exemples sont :
+
+- le tri par insertion
+- le tri fusion
+
+## Tri par insertion
+
+Note à propos du pseudo-code: le but est d'être précis tout en s'abstrayant des détails quand cela
+permet de mieux comprendre l'algorithme en lui-même. Par exemple, on peut remplacer certaines
+instructions par une phrase en langue naturelle.
+
+Le code peut être trouvé à peu près n'importe où. Il consiste à chaque tour de boucle à
+prendre le prochain élément non trié, à faire glisser tous les éléments qui lui sont supérieurs
+vers la fin du tableau et finalement l'insérer juste après le premier élément inférieur.
+
+### La notion d'invariant de boucle
+
+Un invariant de boucle est une propriété qui est vraie avant de commencer chaque itération de boucle
+(comprenant l'invalidation de la condition de boucle). Trois propriété à montrer:
+
+- vrai à l'initialisation,
+- si vrai au rang i, vrai au rang i+1,
+- vrai à la terminaison.
```

Comment:
Peut-être qu'il faudrait dire que c'est lorsque la condition d'arrêt est invalidée (ou fausse) ? Les invariants de boucles sont indépendants de la terminaison il me semble. Un invariant peut-être vraie pour des boucles qui ne terminent pas.

Reply:
Tout à fait. Peut être que préciser que c'est vraiment à la fin de l'itération (et pas deux instructions plus loin), mais ça semble superflus.